unix/vncpasswd/vncpasswd.cxx

```diff
@@ -165,24 +165,40 @@ int main(int argc, char** argv)
     FILE* fp = fopen(fname,"w");
     if (!fp) {
       fprintf(stderr,"Couldn't open %s for writing\n",fname);
+      delete [] fname;
```

Comment:
This is very wrong (UB). fname might be allocated by` strdup` so it mast be freed by `free` in that case, not `delete []`. But it looks like it's also possible to be allocated by `new[]`. This code is some ugly C/C++ mix ;P There are probably issues like this all over the place.

Reply:
It uses new[] all the time, because it uses internally implemented strdup(), which uses new[] as well, so calling delete[] should be just fine.

Reply:
Wow, that's totally not confusing at all. Are you sure about that? Can you point me to this internal implementation, just to be safe (I've found some other similar functions like `strDup`, but no `strdup`)?

Reply:
You are right, I though both are strdup, in that case maybe strDup should be used instead?

Reply:
Looks like it would be the least invasive change (though it wouldn't make current code any better, just more correct).